alkira/resource_alkira_service_checkpoint.go

```diff
@@ -0,0 +1,345 @@
+package alkira
+
+import (
+	"log"
+
+	"github.com/alkiranet/alkira-client-go/alkira"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceAlkiraCheckpoint() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manage checkpoint services",
+		Create:      resourceCheckpoint,
+		Read:        resourceCheckpointRead,
+		Update:      resourceCheckpointUpdate,
+		Delete:      resourceCheckpointDelete,
+
+		Schema: map[string]*schema.Schema{
+			"auto_scale": {
+				Description: "Indicate if `auto_scale` should be enabled for your checkpoint" +
+					"firewall. `ON` and `OFF` are accepted values. `OFF` is the default if " +
+					"field is omitted",
+				Type:         schema.TypeString,
+				Default:      "OFF",
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"ON", "OFF"}, false),
+			},
+			"billing_tag_ids": {
+				Description: "Billing tag IDs to associate with the service.",
+				Type:        schema.TypeList,
+				Optional:    true,
+				Elem:        &schema.Schema{Type: schema.TypeInt},
+			},
+			"cxp": {
+				Description: "CXP region.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"credential_id": {
+				Description: "ID of Checkpoint Firewall credential managed by credential resource.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"description": {
+				Description: "The description of the checkpoint service.",
+				Type:        schema.TypeString,
+				Optional:    true,
+			},
+			"instances": {
+				Type: schema.TypeSet,
+				//Required: true,
+				Optional: true,
+				Description: "An array containing properties for each Checkpoint Firewall instance " +
+					"that needs to be deployed. The number of instances should be equal to " +
+					"`max_instance_count`.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Description: "The name of the Checkpoint Firewall instance.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"credential_id": {
```

Comment:
Do we still need credential ID here? I think the top one should cover this already?